tests/sapi.rs

```diff
@@ -0,0 +1,81 @@
+#![cfg_attr(windows, feature(abi_vectorcall))]
+extern crate ext_php_rs;
+
+#[cfg(feature = "embed")]
+use ext_php_rs::builders::SapiBuilder;
+#[cfg(feature = "embed")]
+use ext_php_rs::embed::{ext_php_rs_sapi_startup, Embed};
+use ext_php_rs::ffi::{
+    php_module_shutdown, php_module_startup, php_request_shutdown, php_request_startup,
+    sapi_shutdown, sapi_startup, ZEND_RESULT_CODE_SUCCESS,
+};
+use ext_php_rs::prelude::*;
+use ext_php_rs::zend::try_catch;
+
+#[test]
+#[cfg(feature = "embed")]
+fn test_sapi() {
+    let builder = SapiBuilder::new("test", "Test");
+    let sapi = builder.build().unwrap().into_raw();
+    let module = get_module();
+
+    unsafe {
+        ext_php_rs_sapi_startup();
+    }
+
+    unsafe {
+        sapi_startup(sapi);
+    }
+
+    unsafe {
+        php_module_startup(sapi, module);
+    }
+
+    let result = unsafe { php_request_startup() };
```

Comment:
Since we separate module startup / request startup we can imagine having a loop here to handle multiple requests / execution with memory cleaned between them